src/GrumPHP/Task/PhpVersion.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * PhpVersion task
+ */
+class PhpVersion implements TaskInterface
+{
+    /**
+     * @var \GrumPHP\Util\PhpVersion
+     */
+    private $util;
+
+    /**
+     * @var GrumPHP
+     */
+    private $grumPHP;
+
+    /**
+     * PhpVersion constructor.
+     * @param GrumPHP $grumPHP
+     * @param \GrumPHP\Util\PhpVersion $util
+     */
+    public function __construct(GrumPHP $grumPHP, \GrumPHP\Util\PhpVersion $util)
+    {
+        $this->grumPHP = $grumPHP;
+        $this->util = $util;
+    }
+
+    /**
+     * This methods specifies if a task can run in a specific context.
+     *
+     * @param ContextInterface $context
+     *
+     * @return bool
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return $context instanceof RunContext;
+    }
+
+    /**
+     * @param ContextInterface $context
+     *
+     * @return TaskResultInterface
+     */
+    public function run(ContextInterface $context)
+    {
+        // Check the current version
+        $config = $this->getConfiguration();
+        if (!$this->util->isSupportedVersion(PHP_VERSION)) {
+            return TaskResult::createFailed(
+                $this,
+                $context,
+                sprintf('PHP version %s is unsupported', PHP_VERSION)
+            );
+        }
+
+        // Check the project version if defined
+        if (array_key_exists('project', $config) !== null) {
```

Comment:
There is no need for `array_key_exists` here. The `OptionsResolver` will make sure it exists since you default it to null.